### PR TITLE
support for dia

### PR DESCRIPTION
**Pull request by Lazarus-931**
this adds dia, a Nari Labs encoder-decoder transformer that byte-tokenizes text and autoregressively decodes 9 delayed DAC codebooks with classifier-free guidance, decoded to audio by the Descript audio codec.

i numerically validated the encoder, decode-step logits, and full greedy waveform against mlx-audio (~1e-4), plus an end-to-end test synthesizes audio via mlx-community/Dia-1.6B

### CODE REVIEW

**Comment by Lazarus-931** [comment]
@lucasnewman

**Review by lucasnewman — APPROVED** [review]
Looks good, thanks!

**Comment by lucasnewman** [comment]
@Lazarus-931 Looks like the DAC tests are failing.
